Feature_1/security, auth 구현

Ryeolee opened the pull request:
1. gradle 추가
 - implementation 'org.springframework.boot:spring-boot-starter-data-redis'
 -  testImplementation 'it.ozimov:embedded-redis:0.7.2'
 - implementation 'org.springframework.boot:spring-boot-starter-webflux'

2. security 적용
 - 토큰 검증 유무를 configure()와 filterChain()로 나눔
 
3. api 개발
 - 구글, 카카오 로그인
 - 로그아웃 
 - 토큰 재발급

Ryeolee committed:
[feat] 시큐리티 기초 설정 -> application.yml 파일 안읽힘

Ryeolee committed:
[feat] 1. 구글, 카카오 로그인 완료
       2. gradle(webflux, redis) 추가

Ryeolee committed:
[feat] 로그아웃 완료

Ryeolee committed:
[feat] 1. 로그인 리팩토링 2. 재발급 api 완료

Ryeolee committed:
[fix] 1. 로그인 SocailLogin 함수 반환값 수정

MinchoGreenT commented:
1. 레디스 관련 정보들을 application.yml에 추가하신 듯 한데, 수정사항이 있다면 노션의 해당 페이지의 코드도 수정 후 언급 부탁드립니다!
2. SocialLogin 클래스의 verifySocialUserData 메서드의 경우 반환값이 사용되지 않는 것으로 보입니다. 유저가 존재하지 않는 경우의 에러를 잡으려고 하신 것 같은데, 맞을까요?

Ryeolee committed:
[fix] 1. SocialLogin login함수 매개변수명 수정
socialUserData2 -> socialUserData

Ryeolee commented:
네~ 소셜에서 유저에 대한 정보를 받아오고 데이터 체크를 위한 함수입니답

Ryeolee committed:
[fix] 에러처리 방식 수정

Ryeolee committed:
[fix] 1. 도메인별 Exception 패키기 생성 및 적용
      2. socialLogin 클래스 validateUserDataAndSignUp -> socialSignUp 함수 변경 orElseThrow 적용

MinchoGreenT commented:
reissueToken 메서드에서 validateRefreshToken 메서드의 반환값이 사용되고 있지 않습니다
-> 토큰에 오류가 있다면 응답을 반환하지만 아무일도 일어나지 않습니다

이부분은 예외 클래스를 하나 만들어서 AuthExceptionHandler로 보내야 할 것 같네요
함수의 반환값을 void로 하고 if문에서 오류가 날 경우 예외를 던지는 식으로요!

그부분만 수정하고 바로 머지해도 될 것 같습니다.

Ryeolee committed:
[fix] 1. validateRefreshToken함수 auth 전역 에러 처리 변경
